src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -166,10 +159,48 @@ public void AddContext<TContext>() where TContext : JsonSerializerContext, new()
         {
             VerifyMutable();
             TContext context = new();
-            _serializerContext = context;
+            _typeInfoResolver = context;
+            _isLockedInstance = true;
             context._options = this;
         }
 
+        /// <summary>
+        /// Gets or sets JsonTypeInfo resolver.
+        /// </summary>
+        public IJsonTypeInfoResolver TypeInfoResolver
+        {
+            [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+            [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+            get
+            {
+                return _typeInfoResolver ?? DefaultJsonTypeInfoResolver.RootDefaultInstance();
```

Comment:
TODO: consider https://github.com/krwq/runtime-1/pull/8

Reply:
let's discuss this separately and if needed let's do this separately from this PR. I personally prefer when you can query options `What's the type info you have used for this type?` similarly to what you can do with `GetConverter`